scripts/apps/master-desk/components/OverviewComponent.tsx

```diff
@@ -58,11 +68,16 @@ export class OverviewComponent extends React.Component<IProps, IState> {
             dataApi.create('desks/all/overview/stages', {
                 filters: this.props.filters,
             }).then((res) => {
-                this.setState({stagesCount: res['_items'], view: 'detailed'});
+                // eslint-disable-next-line react/no-did-update-set-state
+                this.setState({stagesCount: res['_items']});
+
+                if (this.props.currentView === IMasterDeskViews.card) {
+                    this.setView(IMasterDeskViews.detailed);
+                }
             });
-        } else if (this.state.view === 'detailed') {
+        } else if (this.props.currentView === IMasterDeskViews.detailed) {
             // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({view: 'card'});
+            this.setView(IMasterDeskViews.card);
         }
 
         // Filter items by desk name
```

Comment:
A lot of code inside `componentDidUpdate` often means that there's something wrong with the code structure.

A few blocks can probably be dropped, by removing `filteredDesks` from state. It is a value that is derived from props and shouldn't be put in the state. Creating a variable with filtered values in the render method should work fine.

Reply:
I tried to make this quick, but lot of functions for filtering and getting items from server should be moved from Overview to Masterdesk component. Let's keep it like this for now, and in future we can refactorize this. We need this in new release ASAP.